novawallet/Modules/NetworkDetails/NetworkDetailsInteractor.swift

```diff
@@ -0,0 +1,375 @@
+import UIKit
+import SubstrateSdk
+import SoraFoundation
+import Operation_iOS
+
+final class NetworkDetailsInteractor {
+    weak var presenter: NetworkDetailsInteractorOutputProtocol?
+
+    private var chain: ChainModel
```

Comment:
Yes, we can pass the id instead of chain, but we need chain to operate anyway, so if we pass just the id, it will force us to check optional chain in almost every method of the class. 
We're subscribing for chain registry changes so any chain change will be processed anyway.